### PR TITLE
Finishing logic & Adding unit tests

### PR DESCRIPTION
**Pull request by SpaceHexagon**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+        if ( duration > 0 )
+
+            setTimeout( ()=> { this._colorEffect( duration -1, element ) }, 1000 )
+
+    }
+
+    _randomColor ( ) {
+
+        let rgb = [ 0, 0, 0 ].map( c => 20 + Math.floor( Math.random() * 4 ) * 63 )
```
Comment (jbryantveem) [inline src/index.js]: What is the range you expect to find in this formula? (20 + Math.floor( Math.random() * 4 ) * 63). Can this formula generate every random RGB color?

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
+
+class BGCEffect {
+
+    /**
+     * @param {Object} element 
+     */
+    constructor ( element ) {
+
+        this.element = element ? element : (document && document.body) ? document.body : null
```
Comment (jbryantveem) [inline src/index.js]: Will the document or document.body ever not exist? Assuming we don't need to check for document body, is there another way you can write this line without using any ternaries that would be more readable?

File: src/index.js (hunk 1 above)
Comment (SpaceHexagon) [reply src/index.js]: my thinking here was to quantize the the possible colors and reduce the possibility of consecutive colors "looking" similar. I can change this if need be

File: src/index.js (hunk 2 above)
Comment (SpaceHexagon) [reply src/index.js]: It's possible that it wont exist, In which case an error message logs.
I'll post the revision in a sec.

Comment (SpaceHexagon) [reply src/index.js]: https://github.com/SpaceHexagon/background-color-effect/pull/2/commits/c289f94cfd33d1e8b28cbe10beb7e5beac85cbb2

Comment (jbryantveem) [reply src/index.js]: Sorry, I think I should have made my point a bit more clear. Assuming we do *not* need to check for document.body to exist, is there an equivalent 1-line of code we could write that would do the same thing as the ternary: this.element = element ? element : document.body

File: src/index.js (hunk 1 above)
Comment (jbryantveem) [reply src/index.js]: No need to change. I just wanted to understand the reasoning behind the chosen formula.

File: src/index.js (hunk 2 above)
Comment (SpaceHexagon) [reply src/index.js]: I would do ```this.element = element || document.body```, or wrap it in a function call if it were more complex.

**Pull request merged**